src/spec-common/injectHeadless.ts

```diff
@@ -349,12 +370,12 @@ export async function runLifecycleHooks(params: ResolverParameters, lifecycleHoo
 
```

Comment:
Does it matter for performance if the promise is resolved once here? I don't think so but i'm not positive

Reply:
I guess its the same promise so once it solves once it will continue to be resolved :)